plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -358,19 +442,28 @@ private String getVirtualRouterIP(Network network, Collection<String> addressRan
         Set<Long> allIPsInCidr = NetUtils.getAllIpsFromCidr(subnet, cidrSize, new HashSet<Long>());
 
         if (allIPsInCidr.size() > 3) {
+            //get the second IP and see if it the networks GatewayIP
             Iterator<Long> ipIterator = allIPsInCidr.iterator();
             long vip = ipIterator.next();
             if (NetUtils.ip2Long(network.getGateway()) == vip) {
-                s_logger.debug("Gateway of the Network(" + network.getUuid() + ") has the first IP " + NetUtils.long2Ip(vip));
+                if (s_logger.isDebugEnabled()) {
+                    s_logger.debug("Gateway of the Network(" + network.getName() + ") has the first IP " + NetUtils.long2Ip(vip));
+                }
                 vip = ipIterator.next();
                 virtualRouterIp = NetUtils.long2Ip(vip);
-                s_logger.debug("So, reserving the 2nd IP " + virtualRouterIp + " for the Virtual Router IP in Network(" + network.getUuid() + ")");
+                if (s_logger.isDebugEnabled()) {
+                    s_logger.debug("So, reserving the 2nd IP " + virtualRouterIp + " for the Virtual Router IP in Network(" + network.getName() + ")");
```

Comment:
This log message assumes context from the previous message.  This message will likely not appear directly after the previous nor will is it easy to associate with the previous message.  Please change the message to usable without information from the previous message.